utilities/scripts/python/vendor_files_check.py

```diff
@@ -1,11 +1,188 @@
-
 # cspell: words colordiff
 
 # Checks if two files that should exist DO, and are equal.
 # used to enforce consistency between local config files and the expected config locked in CI.
 
-import python.cli as cli
+import python.exec_manager as exec_manager
+import tomllib
+import time
+from dataclasses import dataclass
+
+
+def colordiff_check(
+    vendor_file_path: str, provided_file_path: str
+) -> exec_manager.Result:
+    return exec_manager.cli_run(
+        f"colordiff -Nau {provided_file_path} {vendor_file_path}",
+        name=f"Checking if Provided File {provided_file_path} == Vendored File {vendor_file_path}",
+    )
+
+
+def toml_diff_check(
+    vendor_file_path: str,
+    provided_file_path: str,
+    strict: bool = True,
+    log: bool = True,
+):
+    command_name = (
+        f"{'Strict' if strict else 'Non Strict'} Checking"
+        + f"if Provided File {provided_file_path} == Vendored File {vendor_file_path}"
+    )
+    with open(vendor_file_path, "rb") as vendor_file, open(
+        provided_file_path, "rb"
+    ) as provided_file:
+
+        def procedure() -> exec_manager.ProcedureResult:
+            vendor_obj = tomllib.load(vendor_file)
+            provided_obj = tomllib.load(provided_file)
+
+            diff = (
+                __strict_diff__(vendor_obj, provided_obj)
+                if strict
```

Comment:
```suggestion
                if not strict
```

Reply:
no no, it is a correct one. I am using a python ternary operator ` [on_true] if [expression] else [on_false]`.
So if `strict` is `True` `__strict_diff__` evaluates, otherwise `__inclusion_diff__`

Reply:
Or you are arguing here not use ternary operator ?